### ISSUE TITLE
Convert '==' to 'Equals' and reverse

### ISSUE THREAD

**Issue by ElemarJR** [code-cracker/code-cracker#118]
This:

``` csharp
var areEqual = a == b;
```

Becomes:

``` csharp
var areEqual = a.Equals(b);
```

Category: `Refactoring`
Severity: `Hidden`
Diagnostic ID: CC0085 (converting `==` to `Equals`)
Diagnostic ID: CC0086 (converting `Equals` to `==`)

**Comment by Kavignon** [comment]
Is someone working on this at the moment ?

**Comment by giggio** [comment]
@Kavignon No. But it is not `Ready` yet. We need a test case, and a diagnostic id.

**Comment by giggio** [comment]
@Kavignon Feel free to suggest one. @ElemarJR You might want to add it too?

**Comment by Kavignon** [comment]
I'm not quite sure about this, but I think that it's better to compare string with the == operator than the Equals method. Please tell me if I'm wrong about this 
That could be one of the test cases.
@giggio

**Comment by giggio** [comment]
@Kavignon @ElemarJR I added the test cases. @Kavignon If you want to work on it let us know.
This is a refactoring, the user might want to change one way or the other.

**Comment by Kavignon** [comment]
@giggio where are the test cases ? 
So, if I can it right, I'll have to produce the following

Category: Refactoring
Severity: Hidden
Diagnostic ID: CC0085 (converting == to Equals)
Diagnostic ID: CC0086 (converting Equals to ==)

I'll have to see in the library how refactoring is done. I'm used to analyzing but I've no prior experience with refactoring so this will be a nice experience. 
I'll make a pull request when I'm done !
I don't seem to be able to assign this issue to myself
1. How can I assign this to myself?
2. Can someone assign it to me ?

**Comment by carloscds** [comment]
@Kavignon I will put working in this issue.

**Comment by giggio** [comment]
@Kavignon We cannot assign the issue to you, github only allows us to assign issues to project maintainers (which are part of a team).
Do not create a Roslyn refactoring. We are not able to deploy refactorings with Nuget, so we are not using them. You need to create an analyzer which creates a Hidden diagnostic, this is how we are working refactorings in CC.
If you have any questions let us know.

**Comment by Kavignon** [comment]
I am so sorry. I have been really busy in the past weeks. I'm back on track now and it should not take long before I start making pull requests !

**Comment by giggio** [comment]
Nice! :)

**Comment by giggio** [comment]
@Kavignon How is this coming?
